base_layer/wallet/src/utxo_scanner_service/utxo_scanner_task.rs

```diff
@@ -590,6 +578,18 @@ where
             if EncryptedData::decrypt_data(&recovery_key, &commitment, &encrypted)
                 .ok()
                 .is_some()
+            {
+                found_outputs.push(output.clone());
+                continue;
+            }
+
+            // Change outputs just use the view key.
+            if self
```

Comment:
isn't this already done in the lines above?

Reply:
no, its stealth key check first, then view key check